test/byContract/IexecEscrow/IexecEscrowToken.test.ts

```diff
@@ -0,0 +1,473 @@
+                .to.changeTokenBalances(
+                    iexecPoco,
+                    [...depositForArrayParams.targets],
+                    [...depositForArrayParams.amounts],
+                )
+                .to.emit(iexecPoco, 'Transfer')
+                .withArgs(
+                    AddressZero,
+                    depositForArrayParams.targets[0],
+                    depositForArrayParams.amounts[0],
+                )
+                .to.emit(iexecPoco, 'Transfer')
+                .withArgs(
+                    AddressZero,
+                    depositForArrayParams.targets[1],
+                    depositForArrayParams.amounts[1],
+                );
+        });
+        it('Should not depositForArray with mismatched array lengths', async () => {
+            const depositForArrayParams = {
+                amounts: [standardAmount.mul(2), standardAmount, standardAmount.div(2)],
+                targets: [accountB.address, accountC.address],
+            };
+            const depositForArrayArgs = Object.values(depositForArrayParams) as [
+                BigNumber[],
+                string[],
+            ];
+            const depositTotalAmount = getTotalAmount(depositForArrayParams.amounts);
+
+            await rlcInstanceAsAccountA
+                .approve(iexecPoco.address, depositTotalAmount)
+                .then((tx) => tx.wait());
+            await expect(
+                iexecPocoAsAccountA.depositForArray(...depositForArrayArgs),
+            ).to.be.revertedWith('invalid-array-length');
+        });
+        it('Should not depositForArray with address zero in target', async () => {
+            const depositForArrayParams = {
+                amounts: [standardAmount, standardAmount.mul(2)],
+                targets: [AddressZero, accountB.address],
+            };
+            const depositForArrayArgs = Object.values(depositForArrayParams) as [
+                BigNumber[],
+                string[],
+            ];
+            const depositTotalAmount = getTotalAmount(depositForArrayParams.amounts);
+
+            await rlcInstanceAsAccountA
+                .approve(iexecPoco.address, depositTotalAmount)
+                .then((tx) => tx.wait());
+            await expect(
+                iexecPocoAsAccountA.depositForArray(...depositForArrayArgs),
+            ).to.be.revertedWith('ERC20: mint to the zero address');
+        });
+    });
+
+    describe('Withdraw', () => {
+        it('Should withdraw tokens', async () => {
+            await rlcInstanceAsAccountA
+                .approve(iexecPoco.address, standardAmount)
+                .then((tx) => tx.wait());
+            await iexecPocoAsAccountA.deposit(standardAmount).then((tx) => tx.wait());
+
+            expect(await iexecPocoAsAccountA.callStatic.withdraw(standardAmount)).to.be.true;
+            await expect(iexecPocoAsAccountA.withdraw(standardAmount))
+                .to.changeTokenBalances(
```

Comment:
swap burn & withdraw checks everywhere to maintain chronological order

Reply:
Done on those two [commits](https://github.com/iExecBlockchainComputing/PoCo/pull/141/files/9528dd4715415dbe283c681c960de4182dea40c4..74517e154e629006f01b9837259dffbd05d45ebd)